WordSearchTest.java

```diff
@@ -0,0 +1,54 @@
+package assignment3;
+
+import java.util.HashSet;
+import java.util.Set;
+import org.junit.Test;
+
+public class WordSearchTest {
+  WordSearch wordSearch = new WordSearch();
+
+  @Test
+  public void findAllWords() {
+    char[][] grid = {{'A', 'A', 'R'}, {'T', 'C', 'D'}};
+    HashSet<String> words = new HashSet<>();
+    words.add("CAR");
+    words.add("CARD");
+    words.add("CART");
+    words.add("CAT");
+    Dictionary dictionary = new Dictionary(words);
+
+    Set<String> wordsFound = WordSearch.findAllWords(grid, dictionary);
+
+    assert (wordsFound.contains("CAR"));
+    assert (wordsFound.contains("CAT"));
+    assert (wordsFound.contains("CARD"));
+    assert (!wordsFound.contains("CAP"));

```

Comment:
Good! It is important to not only check that expected words were returned, but also the opposite - that no unexpected words were there.

Stronger versions of this check:

- check for positives (expected words) and then also verify size of the container to make sure that it does not include any more words.
- create a set of expected words and use `assertEquals(wordsExpected, wordsFound)`.